esp/esdllib/esdl_script.cpp

```diff
@@ -2310,6 +2340,318 @@ class CEsdlTransformOperationFunction : public CEsdlTransformOperationWithChildr
     }
 };
 
+/**
+ * @brief Helper structure used by both tx-summary-value and tx-summary-timer.
+ *
+ * There is common data used by both operations. Because one operation is childless and the other
+ * allows children, they cannot have a common base class. Instead each contains an instance of this
+ * structure to provide common handling of the common data.
+ */
+struct TxSummaryCoreInfo
```

Comment:
No need to change, but just mentioning you probably could have made this a templatized base class,.. and then derived each operation from it:

class CEsdlTransformOperationTxSummaryValue : public TxSummaryCoreInfo<CEsdlTransformOperationWithoutChildren>

Reply:
Right.  Didn't look at the details of the constructors.  Would be too complicated and I would have probably complained ;)